File: 1. Two Sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> map = new HashMap<>();
        List<Integer> result = new ArrayList<>();
        for (int i = 0; i < nums.length; i++) {
            if (map.containsKey(target - nums[i])) {
                result.add(map.get(target - nums[i]));
                result.add(i);
            } else {
                map.put(nums[i], i);
            }
        }
        return result.stream().mapToInt(i -> i).toArray();
    }
}

File: 1. Two Sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        m = {}
        result = []
        for i in range(len(nums)):
            if target - nums[i] not in m:
                m[nums[i]] = i
            else:
                result.append(m[target - nums[i]])
                result.append(i)
        return result
